engine_test.go

```diff
@@ -103,3 +103,35 @@ func TestWithRequestContentType(t *testing.T) {
 		require.False(t, ok)
 	})
 }
+
+func TestWithResponseContentType(t *testing.T) {
+	t.Run("base", func(t *testing.T) {
+		e := NewEngine()
+		require.NotNil(t, e.responseContentTypes)
+		assert.Equal(t, []string{"application/json", "application/xml"}, e.responseContentTypes)
+	})
+
+	t.Run("input", func(t *testing.T) {
+		arr := []string{"application/yml"}
+		e := NewEngine(WithResponseContentType("application/yml"))
+		require.ElementsMatch(t, arr, e.responseContentTypes)
+	})
+
+	t.Run("ensure applied to route", func(t *testing.T) {
+		s := NewServer(WithEngineOptions(
+			WithResponseContentType("application/yml")),
+		)
+		route := Post(s, "/test", dummyController)
+
+		response := route.Operation.Responses.Map()["200"]
+		require.NotNil(t, response)
+		content := response.Value.Content
+		require.NotNil(t, content["application/yml"])
```

Comment:
A contains would provide a clearer error message

```suggestion
		require.Contains(t, content, "application/yml")
```